receiver/awscloudwatchreceiver/logs.go

```diff
@@ -0,0 +1,335 @@
+	defer l.wg.Done()
+
+	t := time.NewTicker(l.pollInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-l.doneChan:
+			return
+		case <-t.C:
+			err := l.pollForLogs(ctx, logGroup, pc)
+			if err != nil {
+				l.logger.Error("there was an error during the poll", zap.String("log group", logGroup), zap.Error(err))
+			}
+		}
+	}
+}
+
+func (l *logsReceiver) pollForLogs(ctx context.Context, logGroup string, pc pollConfig) error {
+	err := l.ensureSession()
+	if err != nil {
+		return err
+	}
+	nextToken := aws.String("")
+	for nextToken != nil {
+		select {
+		// if done, we want to stop processing paginated stream of events
+		case _, ok := <-l.doneChan:
+			if !ok {
+				return nil
+			}
+		default:
+			startTime := time.Now().Add(-l.pollInterval)
+			endTime := time.Now()
+			input := pc.request(l.maxEventsPerRequest, &startTime, &endTime)
+			resp, err := l.client.FilterLogEventsWithContext(ctx, input)
+			if err != nil {
+				l.logger.Error("unable to retrieve logs from cloudwatch", zap.String("log group", logGroup), zap.Error(err))
+				break
+			}
+			observedTime := pcommon.NewTimestampFromTime(time.Now())
+			logs := l.processEvents(observedTime, logGroup, resp)
+			if logs.LogRecordCount() > 0 {
+				if err = l.consumer.ConsumeLogs(ctx, logs); err != nil {
+					l.logger.Error("unable to consume logs", zap.Error(err))
+					break
+				}
+			}
+			nextToken = resp.NextToken
+		}
+	}
+	return nil
+}
+
+func (l *logsReceiver) processEvents(now pcommon.Timestamp, logGroupName string, output *cloudwatchlogs.FilterLogEventsOutput) plog.Logs {
+	logs := plog.NewLogs()
+	for _, e := range output.Events {
+		if e.Timestamp == nil {
+			l.logger.Error("unable to determine timestamp of event as the timestamp is nil")
+			continue
+		}
+
+		if e.Message == nil {
+			l.logger.Error("no message was present on the event", zap.Any("event.id", e.EventId))
+			continue
+		}
+
+		rl := logs.ResourceLogs().AppendEmpty()
+		resourceAttributes := rl.Resource().Attributes()
+
+		safePutAttribute("aws.region", &l.region, resourceAttributes.PutString)
+		safePutAttribute("cloudwatch.log.group.name", &logGroupName, resourceAttributes.PutString)
+		safePutAttribute("cloudwatch.log.stream", e.LogStreamName, resourceAttributes.PutString)
+
+		logRecord := rl.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+		logRecord.SetObservedTimestamp(now)
+		ts := time.UnixMilli(*e.Timestamp)
+		logRecord.SetTimestamp(pcommon.NewTimestampFromTime(ts))
+
+		logRecord.Body().SetStr(*e.Message)
+		safePutAttribute("id", e.EventId, logRecord.Attributes().PutString)
+	}
+	return logs
+}
+
+func (l *logsReceiver) discoverGroups(ctx context.Context, auto *AutodiscoverConfig) ([]namesPollConfig, []prefixPollConfig, error) {
+	l.logger.Debug("attempting to discover log groups.", zap.Int64("limit", auto.Limit))
+
+	newNamedPolls := []namesPollConfig{}
+	newPrefixedPolls := []prefixPollConfig{}
```

Comment:
Maybe it's just me but I find this "polls" naming pattern kind of awkward. It puts the focus on a generic aspect of out the logs are collected (polling), rather than the meaningful way in which we are trying to manage them (groups).

Would it make sense to refer to these as `groupNames` and `groupPrefixes`, or similar?